planning/behavior_path_planner/src/utils/goal_planner/default_fixed_goal_planner.cpp

```diff
@@ -38,12 +40,45 @@ BehaviorModuleOutput DefaultFixedGoalPlanner::plan(
   return output;
 }
 
+lanelet::ConstLanelets DefaultFixedGoalPlanner::extractLaneletsFromPath(
+  const PathWithLaneId & refined_path,
+  const std::shared_ptr<const PlannerData> & planner_data) const
+{
+  const auto & rh = planner_data->route_handler;
+  lanelet::ConstLanelets refined_path_lanelets;
+  for (size_t i = 0; i < refined_path.points.size(); ++i) {
+    const auto & path_point = refined_path.points.at(i);
+    int64_t lane_id = path_point.lane_ids.at(0);
+    lanelet::ConstLanelet lanelet = rh->getLaneletsFromId(lane_id);
+    bool is_unique =
+      std::find(refined_path_lanelets.begin(), refined_path_lanelets.end(), lanelet) ==
+      refined_path_lanelets.end();
+    if (is_unique) {
+      refined_path_lanelets.push_back(lanelet);
+    }
+  }
+  return refined_path_lanelets;
+}
+
+bool DefaultFixedGoalPlanner::isPathValid(
+  const PathWithLaneId & refined_path,
+  const std::shared_ptr<const PlannerData> & planner_data) const
+{
+  const lanelet::ConstLanelets lanelets = extractLaneletsFromPath(refined_path, planner_data);
+  // std::any_of detects whether any point lies outside lanelets
+  bool anyPointOutsideLanelets = std::any_of(
+    refined_path.points.begin(), refined_path.points.end(),
+    [&lanelets](const auto & refined_path_point) {
+      return !utils::isInLanelets(refined_path_point.point.pose, lanelets);
+    });
+  return !anyPointOutsideLanelets;  // NOT anyPointOutsideLanelets --> isPathValid
```

Comment:
you probably dont have to put the comments `// NOT anyPointOutsideLanelets --> isPathValid`.

Reply:
deleted to enhance the readability [https://github.com/autowarefoundation/autoware.universe/commit/605174e6e720e97f17c5ae93d130d9c22f3366c5](https://github.com/autowarefoundation/autoware.universe/commit/605174e6e720e97f17c5ae93d130d9c22f3366c5)